### PR TITLE
Restore anonId to dedicated ExPlat anonId

### PR DESCRIPTION
**Pull request by claudiucelfilip**
Related to #

## Proposed Changes

This is variation of the #82269 where instead of restoring the `tk_ai` we store it in `tk_ai_explat`. Together with the changes to the ExPlat assigner (D123644-code), the assignment should be persisted across the login barrier.

## Testing Instructions

* Visit a Logged Out Landing page, so that the `tk_ai` cookie is populated
* Go through the Log in flow and the Signup flow and make sure there's a `tk_ai_explat` cookie with the initial `tk_ai`.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by aaronyan — COMMENTED** [review]
Amazing work @claudiucelfilip, thank you for making this happen!

Do we need to remove the `tk_ai_explat` token on log-out (passive and active)? I think this can live on forever, since it identifies the user and their browser and we want them to get the same assignment variation even if they log out and log in with a different account. Thoughts?

**Review by dzver — APPROVED** [review]
LGTM

Given that this seems to be the preferred approach by Data folks, happy to approve the main PR after this is merged into it as well.

**Pull request merged**
